Horizon/Horizon/Sources/Features/Inbox/HInboxAssembly.swift

```diff
@@ -23,16 +23,32 @@ struct HInboxAssembly {
     static func makeView() -> UIViewController {
         let inboxViewModel = HInboxViewModel()
         let viewModel = HEmbeddedWebPageContainerViewModel(
-            webPage: inboxViewModel
+            webPage: inboxViewModel,
+            navigationDelegate: inboxViewModel,
+            javaScript: buttonTappedJS,
+            observeKey: "buttonTapped"
         )
 
         let viewController = CoreHostingController(
             HEmbeddedWebPageContainerView(
                 viewModel: viewModel
             )
         )
-        let nav = CoreNavigationController(rootViewController: viewController)
-        nav.navigationBar.useGlobalNavStyle()
         return viewController
     }
+
+    private static var buttonTappedJS: String {
+         """
+        document.addEventListener('click', function(event) {
+            var element = event.target;
+            while (element) {
+                if (element.tagName === 'BUTTON') {
+                    window.webkit.messageHandlers.buttonTapped.postMessage(element.innerText);
+                    break;
+                }
+                element = element.parentElement;
+            }
+        });
+        """
+    }
```

Comment:
This will catch all the button events so there's no way to navigate inside the webview. 

https://github.com/user-attachments/assets/419cab6e-3aa0-495a-8ecd-aaf2c9a6e123